src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -507,6 +505,31 @@ public Response updateNeedsReviewDataSets(@QueryParam("dataSetId") Integer dataS
         }
     }
 
+    @PUT
+    @Produces("application/json")
+    @RolesAllowed(ADMIN)
+    @Path("/{id}/datause")
+    public Response updateDatasetDataUse(@Auth AuthUser authUser, @PathParam("id") Integer id, String dataUseJson) {
+        try {
+            User user = userService.findUserByEmail(authUser.getEmail());
+            Gson gson = new Gson();
+            DataUse dataUse = gson.fromJson(dataUseJson, DataUse.class);
+            Dataset originalDataset = datasetService.findDatasetById(id);
+            if (Objects.isNull(originalDataset)) {
+                throw new NotFoundException("Dataset not found: " + id);
+            }
+            if (Objects.equals(dataUse, originalDataset.getDataUse())) {
+                return Response.notModified().entity(unmarshal(originalDataset)).build();
```

Comment:
oh cool, i didn't know you could do more than `ok` - side note, you don't need to use `unmarshal` here, in fact we could completely remove it since we're using Gson by default (same goes for the `.ok` call below)

Reply:
Thanks, that's great, this is what copy paste gets me.